optuna/samplers/_tpe/multi_objective_sampler.py

```diff
@@ -289,14 +290,12 @@ def _split_mo_observation_pairs(
             }
             weights_below = self._calculate_weights_below(lvals, indices_below)
             attrs["weights_below"] = weights_below.tolist()
-            study._storage.set_trial_system_attr(trial._trial_id, _SPLITCACHE_KEY, attrs)
+            self._split_cache[trial._trial_id] = attrs
 
         below = cvals[indices_below]
-        study._storage.set_trial_system_attr(
-            trial._trial_id,
-            _WEIGHTS_BELOW_KEY,
-            [w for w, v in zip(weights_below, below) if v is not None],
-        )
+        self._weights_below[trial._trial_id] = [
+            w for w, v in zip(weights_below, below) if v is not None
+        ]
```

Comment:
It's simpler to cast this as `np.ndarray` here. 
```suggestion
        self._weights_below[trial._trial_id] = np.asarray([
            w for w, v in zip(weights_below, below) if v is not None
        ], dtype=float)
```